### PR TITLE
vdk-core: fix flakey test in test checking logs output

### PR DESCRIPTION
**Pull request by murphp15**
# Why?
We want to fix intermittent failures in tests. Specifically the described in https://github.com/vmware/versatile-data-kit/issues/1177.

# What?
The underlying issue was that `test_log_plugin` was modifying the log level to warning. 
the test `test_run_dbapi_connection_no_such_db_type` relies on it being INFO to pass. 
There are two fixes to prevent this in the future

1. the log level is explicitly set at the start of the test `test_run_dbapi_connection_no_such_db_type`. 
2. in the test `test_log_plugin` we revert the start back to info at the end of it. 

# How has this been tested?
Before my fix I was getting an error reasonably frequently during runs. 
After I added my fix in I ran the pipeline many more times and never encountered the error again. 

# What type of change are you making?
Bug fix (non-breaking change which fixes an issue) or a cosmetic change/minor improvement

Signed-off-by: murphp15 murphp15@tcd.ie

### CODE REVIEW

File: projects/vdk-core/tests/functional/run/test_run_sql_queries.py [hunk 1]
```diff
@@ -75,6 +75,8 @@ def db_connection_recover_operation(self, recovery_cursor: RecoveryCursor) -> No
 
 @mock.patch.dict(os.environ, {})
 def test_run_dbapi_connection_no_such_db_type():
+    logging.getLogger().setLevel(logging.INFO)
```
Comment (ivakoleva) [inline projects/vdk-core/tests/functional/run/test_run_sql_queries.py:78]: why the flakiness, was the test experiencing intermittent issues on python3.10 only?

File: projects/vdk-core/tests/functional/run/test_run_sql_queries.py [hunk 2]
```diff
@@ -75,6 +75,8 @@ def db_connection_recover_operation(self, recovery_cursor: RecoveryCursor) -> No
```
Comment (ivakoleva) [inline projects/vdk-core/tests/functional/run/test_run_sql_queries.py:75]: take a look at the `CONTRIBUTING.md` to find the git commit template format,

**Review by ivakoleva — APPROVED** [review]

File: projects/vdk-core/tests/functional/run/test_run_sql_queries.py (hunk 1 above)
Comment (murphp15) [reply projects/vdk-core/tests/functional/run/test_run_sql_queries.py:78]: I was able to reproduce on python 3.8 too. 
I don't think it was related to python 3.10

File: projects/vdk-core/tests/functional/run/test_run_sql_queries.py (hunk 2 above)
Comment (murphp15) [reply projects/vdk-core/tests/functional/run/test_run_sql_queries.py:75]: Done

**Comment by tozka** [comment]
We currently do not support really contributions from forks 
because Gitlab CI is not triggered for forks (until https://github.com/vmware/versatile-data-kit/issues/854)

So you need to 
1. I will now add you to list of maintainers 
2. Create a PR from a branch.

**Comment by murphp15** [comment]
> We currently do not support really contributions from forks because Gitlab CI is not triggered for forks (until #854)
> 
> So you need to
> 
> 1. I will now add you to list of maintainers
> 2. Create a PR from a branch.


Cool. How long does it take to add me to the list of maintainers?

**Pull request closed without merging**